_posts/2018-08-15-JS-SDK.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: The JS SDK hits version 2.0
+date: 2018-08-15 15:16 -0400
+comments: true
+author: flovilmart
+categories: [announcements, javascript, sdk]
+---
+
+The summer has been really busy and we've dusted out the JS SDK and striped large parts of it's heritage. `Parse.Promises` and `Backbone` style callbacks are definitely out.
+
+<!-- more -->
+
+## *Why the move now you may ask?*
+
+Well, we're also working hard on parse-server version 3.0, which revamps cloud code interface. The Cloud Code 'callbacks' have always been in the form: `response.success('yay');` and `response.error('oops');`.
+
+The `response` object is following the backbone style callback form, with `success` and `error` callbacks. The next generation of Cloud Code will get rid of this interface that belongs to another age. In 2018, `async` functions and `Promise` are part of the JavaScript language, are available in the LTS node runtime, and not only the leading edge.
+
+## *What should I do?*
+
+- If you're not using `Parse.Promise`'s with the JavaScript SDK, your code will break. You'll need to rewrite your code from the backbone style success/error callbacks to use promises.
+
+- If you're using `Parse.Promise`'s the migration should be minimal. `Parse.Promise` was compatible to a certain extent with the native `Promise` interface.
```

Comment:
If you're using `Parse.Promise`'s the migration should be painless. `Parse.Promise` was mostly compatible with the native `Promise` interface making the transition pretty straightforward.